File: first_primes/first_primes.go
/*Runtime 0.6s with Go 1.2*/
package main
import "fmt"

func main() {
  var primes [10000]int
  var x int = 2
  for length := 0; length < 10000; x++ {
    isPrime := true
    for i := 0; i < length; i++ {
      if x % primes[i] == 0 {
        isPrime = false
        break
      }
    }
    if isPrime {
      primes[length] = x
      length = length + 1
    }
  }
  fmt.Println(primes)
}

File: first_primes/first_primes.rb
# runtime 4.8s with ruby 2.1.0p0, 4.0s with ruby 2.2.0p0
primes = []
x = 1
begin
  x += 1
  primes << x unless primes.any? { |prime| x % prime == 0 }
end until primes.length == 10000
puts primes.join(' ')
